### DISCUSSION TITLE
cumulative total for recurring task

### DISCUSSION THREAD

**Discussion by gour** [GothenburgBitFactory/timewarrior#597] [Q&A] [answered]
Hello,

I've some (taskwarrior) tasks which are recurring daily - e.g. daily reading or exercising, and I'd like to be able to mark them as complete. Moreover, I want to  track how long I spent on the task each day, but still get a monthly report on how much time I did e.g. read or exercised in total?

Is something like that achievable with task/timewarrior?

**Comment by lauft** [comment]
When you give your recurring tasks a distinctive tag, e.g.  `TRAINING`/`READING`, those will be used for the time tracking in Timewarrior as well when using the `on-modify.timewarrior`hook. At the end of the month you can use the `totals` report to get cumulative overview on your spent time with
```
timew totals :month
```
or for a specific month:
```
timew totals january
```

**Comment by gour** [comment]
> When you give your recurring tasks a distinctive tag, e.g. `TRAINING`/`READING`, those will be used for the time tracking in Timewarrior as well when using the `on-modify.timewarrior`hook. At the end of the month you can use the `totals` report to get cumulative overview on your spent time with

That sounds great...

> ```
> timew totals :month
> ```
> 
> or for a specific month:
> 
> ```
> timew totals january
> ```

...but I wonder what do I miss?

```
$ timew totals
'totals' is not a timew command. See 'timew help'.
 $ timew diag

timew 1.7.1
       Platform: Linux

Compiler:
        Version: 13.2.0
           Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
     Compliance: C++17

Build Features
          Built: Feb 25 2024 09:25:40
          CMake: 3.28.3
     Build type: None

Configuration
  TIMEWARRIORDB: -
            Cfg: /home/gour/.config/timewarrior/timewarrior.cfg (lrw- 0 bytes)
       Database: /home/gour/.local/share/timewarrior (drwx 8 bytes)
        $EDITOR: vim
    Color theme: Built-in default
                 00 01 02 03 04 05 06 07 08 09 10 11 12

Extensions
       Location: /home/gour/.config/timewarrior/extensions (drwx 0 bytes)
               (None)
```

I'm on Debian/Sid(uction)

**Reply by lauft** [reply, marked as answer]
I was missing to mention that you have to install the `totals` extension. You should find it in `/usr/local/share/doc/timew/ext/` directory. Copy it to your extensions directory (`.config/timewarrior/extensions`) and make it runable.
